api/services/namespace.go

```diff
@@ -180,28 +180,18 @@ func (s *service) fillMembersData(ctx context.Context, members []models.Member)
 	return members, nil
 }
 
-// EditNamespace edits the namespace name.
-//
-// It receives a context, used to "control" the request flow,  tenant ID from models.Namespace and the new name to
-// namespace. Name is set to lowercase.
-//
-// EditNamespace returns a models.Namespace and an error. When error is not nil, the models.Namespace is nil.
-func (s *service) EditNamespace(ctx context.Context, tenantID, name string) (*models.Namespace, error) {
-	namespace, err := s.store.NamespaceGet(ctx, tenantID)
-	if err != nil {
-		return nil, NewErrNamespaceNotFound(tenantID, err)
+// EditNamespace edits the namespace with the specified tenant.
+func (s *service) EditNamespace(ctx context.Context, req *requests.NamespaceEdit) (*models.Namespace, error) {
+	changes := &models.NamespaceChanges{
+		Name:          strings.ToLower(req.Name),
+		SessionRecord: req.Settings.SessionRecord,
 	}
 
-	name = strings.ToLower(name)
-	if ok, err := s.validator.Struct(&models.Namespace{Name: name}); !ok || err != nil {
-		return nil, NewErrNamespaceInvalid(err)
-	}
-
-	if namespace.Name == name {
-		return nil, NewErrNamespaceDuplicated(nil)
```

Comment:
I think we should check if already exist a namespace with the desired name.

Reply:
In this context, this namespace was retrieved with the tenantId, which implies (to my understanding) that this snippet is not checking for duplications, but only verifying if the desired name matches the current name.

I understand and agree that the code needs to ensure a unique name. However, since the current state of the code does not accomplish that, I would prefer to simply remove this "useless" snippet, as MongoDB already handles situations where the name is the same.